p2p/server.go

```diff
@@ -200,24 +183,29 @@ func makeBasicHost(ctx context.Context, priv crypto.PrivKey, pub crypto.PubKey,
 		return addrs
 	}
 
+	var idht *dht.IpfsDHT
 	opts := []libp2p.Option{
-		libp2p.Peerstore(ps),
-		libp2p.ListenAddrStrings(fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", listenPort)),
 		libp2p.Identity(priv),
+		libp2p.ListenAddrStrings(fmt.Sprintf("/ip4/0.0.0.0/tcp/%d", listenPort)),
+		libp2p.Security(secio.ID, secio.New),
+		// support any other default transports (TCP)
+		libp2p.Transport(tcp.NewTCPTransport),
+		// Attempt to open ports using uPNP for NATed hosts.
+		libp2p.NATPortMap(),
```

Comment:
I think it is good to enable the NATPortMap by default but we might should give the option to also deactivate it with a config variable. (If you know uPNP is deactivated on your router or the node is running on a public ip address)

Reply:
It maybe would be nice to make the libp2p changes in a seperated PR?

Reply:
We would need the node to be communicated by the external peers and we try to do that through config by explicitly mentioning an external. Having a config to disable this would defeat the purpose IMO. 

Moving libp2p changes to a different PR is more work than it is now since we  have to specifically downgrade some libs and hard to determine. Feels not worth that time IMO

Reply:
Yes, it is in our interest to achieve communication with other peers if we are behind a NAT and uPNP would enable the port forwarding on the router with uPNP automatically. If I understood the `NATPortMap` option correctly.  

However, I thought if you know uPNP is anyway deactivated on you router it doesn't help you.
In IPFS you can deactivate the `NATPortMap` but it is enabled by default but i guess it is also fine to always enable it. 👍 
> Moving libp2p changes to a different PR is more work than it is now since we have to specifically downgrade some libs and hard to determine. Feels not worth that time IMO

makes sense 👍

Reply:
> In IPFS you can deactivate the NATPortMap but it is enabled by default but i guess it is also fine to always enable it.
if lib unable to punch a hole through NAT, we have to ensure we pass the external IP through config, which we do that anyway. So leaving it enabled would be good in case the config has a missing external IP and NAT port mapping is supported